astropy/modeling/functional_models.py

```diff
@@ -862,6 +862,61 @@ def _parameter_units_for_data_units(self, inputs_unit, outputs_unit):
                             ('amplitude', outputs_unit['y'])])
 
 
+class Beta1D(Fittable1DModel):
+    """
+    1-D Lorentz model with a varying power law, a.k.a, beta model.
+
+    Parameters
+    ----------
+    amplitude : float
+        Amplitude at x=xpos
+    beta : float
+        Beta index
+    r0 : float
+        Core radius
+    xpos : float
+        Offset from x=0
+
+    Notes
+    -----
+    Model formula:
+
+        .. math:: f(x) = a * (1 + [(x - x_{pos})/r_0] ^ 2)^(-3 * \\beta + 1/2)
+
+    See Also
+    --------
+    Lorentz1D
+    """
+
+    amplitude = Parameter(default=1)
+    beta = Parameter(default=1)
+    r0 = Parameter(default=1)
+    xpos = Parameter(default=0)
+
+    @staticmethod
+    def evaluate(x, amplitude, beta, r0, xpos):
+        return amplitude * (1 + ((x - xpos) / r0) ** 2) ** (-3 * beta + .5)
+
+    @staticmethod
+    def fit_deriv(x, amplitude, beta, r0, xpos):
+        d_r0 = (-2 * amplitude * ((x - xpos) / r0) ** 2 / r0 * (-3 * beta + .5)
+                * (1 + ((x - xpos) / r0) ** 2) ** (-3 * beta - .5))
+        d_beta = (- 3 * amplitude * (1 + ((x - xpos) / r0) ** 2) ** (-3 * beta
+                  + .5) * np.log(1 + ((x - xpos) / r0) ** 2))
+        d_xpos = (-2 * amplitude * (-3 * beta + .5) * (1 + ((x - xpos) / r0)
+                  ** 2) ** (-3 * beta - .5) * (x - xpos) / r0 ** 2)
+        d_amplitude = (1 + ((x - xpos) / r0) ** 2) ** (-3 * beta + .5)
+
```

Comment:
The same terms are computed many times. For those that makes sense could you compute them once at the beginning and reuse that.